# Equipo 5
- Luis David Davila Torres
- Josue Alonso Salazar Serrato

# Analisando las ventas de videojuegos en el Mundo 
este texto fue editado por Josue

## Descripción 
Construir un modelo que pueda predecir las ventas de videojuegos en función de otras características del conjunto de datos de las Ventas de videojuegos.


## Preguntas de interés.

1.	¿Cuál sería la venta esperada de cierto videojuego con ciertas características si se lanza al mercado por zona específica o en forma global?
2.	Como gerente de una de las compañías de consolas ¿me convendría invertir en una desarrolladora exclusiva enfocada en un mercado para poder generar ingresos en un mercado nuevo, conociendo ya las preferencias de género que existen?

## Introducción
La base de datos está orientada a las ventas mundiales en distintas regiones de videojuegos, esta informacón es bastante útil para generar una predicción o estimación para el lanzamiento de un videojuego con ciertas características en las distintas regiones correspondientes, también puede ayudar en base a las ventas en distintas regiones a saber el impacto de cierto videojuego y si convendría o no su lanzamiento en dada región debido a su volumen de venta y crítica. 


# IMPORTAR LIBRERIAS Y BASE DE DATOS Y VISUALIZAR LOS DATOS

In [1]:
import pandas as pd 
import numpy as np
import matplotlib as plt
import matplotlib.pyplot as plt

%matplotlib inline
SALES = pd.read_csv("C:/Users/jos/josue/Mineria_de_Datos/Avance PIA/BASE USAR.csv", index_col=0)
Data=pd.DataFrame(SALES)



### Visualización de los datos

In [10]:
SALES.head()

Platform  Year_of_Release         Genre Publisher  \
Name                                                                         
Wii Sports                    Wii           2006.0        Sports  Nintendo   
Super Mario Bros.             NES           1985.0      Platform  Nintendo   
Mario Kart Wii                Wii           2008.0        Racing  Nintendo   
Wii Sports Resort             Wii           2009.0        Sports  Nintendo   
Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing  Nintendo   

                          NA_Sales  EU_Sales  JP_Sales  Other_Sales  \
Name                                                                  
Wii Sports                   41.36     28.96      3.77         8.45   
Super Mario Bros.            29.08      3.58      6.81         0.77   
Mario Kart Wii               15.68     12.76      3.79         3.29   
Wii Sports Resort            15.61     10.93      3.28         2.95   
Pokemon Red/Pokemon Blue     11.27      8.89     10.22         1.00   

                          Global_Sales  Critic_Score  Critic_Count User_Score  \
Name                                                                            
Wii Sports                       82.53          76.0          51.0          8   
Super Mario Bros.                40.24           NaN           NaN        NaN   
Mario Kart Wii                   35.52          82.0          73.0        8.3   
Wii Sports Resort                32.77          80.0          73.0          8   
Pokemon Red/Pokemon Blue         31.37           NaN           NaN        NaN   

                          User_Count Developer Rating  
Name                                                   
Wii Sports                     322.0  Nintendo      E  
Super Mario Bros.                NaN       NaN    NaN  
Mario Kart Wii                 709.0  Nintendo      E  
Wii Sports Resort              192.0  Nintendo      E  
Pokemon Red/Pokemon Blue         NaN       NaN    NaN

In [11]:
SALES.shape

(16719, 15)

Visualiamos los primeros 5 registros de la base de datos donde vemos algunos nulos

<b>Numero de Filas</b> = 16719   

<b>Numero de Columnas</b>  = 15

### Tipos de Datos por Columna

In [14]:
Data.dtypes

Platform            object
Year_of_Release    float64
Genre               object
Publisher           object
NA_Sales           float64
EU_Sales           float64
JP_Sales           float64
Other_Sales        float64
Global_Sales       float64
Critic_Score       float64
Critic_Count       float64
User_Score          object
User_Count         float64
Developer           object
Rating              object
dtype: object

   <b>Tipo de dato de cada columna</b>
   - Platform         =   object
   - Year_of_Release  =   float64
   - Genre            =    object
   - Publisher        =    object
   - NA_Sales         =   float64
   - EU_Sales         =   float64
   - JP_Sales         =   float64
   - Other_Sales      =   float64
   - Global_Sales     =   float64
   - Critic_Score     =   float64
   - Critic_Count     =   float64
   - User_Score       =    object
   - User_Count       =   float64
   - Developer        =    object
   - Rating           =    object

Tenemos datos flotantes y objeto de las 15 columnas.
  
  <b>Float64=9 </b>
  
  <b>Object= 6 </b>




# Datos Nulos

In [16]:
Data.isnull().any()

Platform           False
Year_of_Release     True
Genre               True
Publisher           True
NA_Sales           False
EU_Sales           False
JP_Sales           False
Other_Sales        False
Global_Sales       False
Critic_Score        True
Critic_Count        True
User_Score          True
User_Count          True
Developer           True
Rating              True
dtype: bool

#### ***

Tenemos 6 Columnas sin datos nulos y 9 con datos nulos

<b> Datos nulos: 9 Columnas </b>

<b> Sin datos nulos: 6 Columnas </b>


### Cantidad de Datos Nulos por Columna

In [17]:
Data.isnull().sum()

Platform              0
Year_of_Release     269
Genre                 2
Publisher            54
NA_Sales              0
EU_Sales              0
JP_Sales              0
Other_Sales           0
Global_Sales          0
Critic_Score       8582
Critic_Count       8582
User_Score         6704
User_Count         9129
Developer          6623
Rating             6769
dtype: int64


<b>Numero de nulos en cada columna</b>
   - Platform         =      0
   - Year_of_Release  =    269
   - Genre            =      2
   - Publisher        =     54
   - NA_Sales         =      0
   - EU_Sales         =      0
   - JP_Sales         =      0
   - Other_Sales      =      0
   - Global_Sales     =      0
   - Critic_Score     =   8582
   - Critic_Count     =   8582
   - User_Score       =   6704
   - User_Count       =   9129
   - Developer        =   6623
   - Rating           =   6769

## Limpieza 

#### Renombramos las columnas 

In [18]:
Data2=Data.rename(columns = {'Platform':'Plataforma','Year_of_Release':'Lanzamiento','Genre':'Genero','NA_Sales':'V_NA',
                             'EU_Sales':'V_EU','JP_Sales':'V_JP','Other_Sales':'V_Otros','Global_Sales':'V_Global',
                             'Critic_Score':'Score','User_Score':'Score_Usuarios','Rating':'ESRB'})

#### Eliminamos Columnas innecesarias 

In [22]:
Data3=Data2
Data3V1=Data3.drop(['Publisher','Critic_Count', 'User_Count', 'Developer','Score_Usuarios'],axis=1) #Sin score de usuarios
Data4 = Data3V1.dropna(axis=0) #Eliminamos los filas que contienen Nulos

#### Se eliminaron 5 columnas:

*Publisher

*Critic Count

*User Count

*Developer 

*Score Usuarios

### Visualización de los datos ya modificados

In [21]:
Data4.head()

Plataforma  Lanzamiento    Genero   V_NA   V_EU  V_JP  \
Name                                                                          
Wii Sports                   Wii       2006.0    Sports  41.36  28.96  3.77   
Mario Kart Wii               Wii       2008.0    Racing  15.68  12.76  3.79   
Wii Sports Resort            Wii       2009.0    Sports  15.61  10.93  3.28   
New Super Mario Bros.         DS       2006.0  Platform  11.28   9.14  6.50   
Wii Play                     Wii       2006.0      Misc  13.96   9.18  2.93   

                       V_Otros  V_Global  Score ESRB  
Name                                                  
Wii Sports                8.45     82.53   76.0    E  
Mario Kart Wii            3.29     35.52   82.0    E  
Wii Sports Resort         2.95     32.77   80.0    E  
New Super Mario Bros.     2.88     29.80   89.0    E  
Wii Play                  2.84     28.92   58.0    E

<b>Después de realizar toda la limpieza nos da un resultado de </b>  7902   <b> Registros y </b>   10 <b> Columnas listos para realizar nuestros análisis </b>

## Descripción estadística *** No se entrega esta parte

In [54]:
Data4['Score'].describe()
#Data4['Genero'].describe()
#Data4['Plataforma'].describe()
#Data4['V_Global'].describe()
#Data4['Score_Usuarios'].unique


count    7902.000000
mean       68.995318
std        13.934261
min        13.000000
25%        60.000000
50%        71.000000
75%        79.000000
max        98.000000
Name: Score, dtype: float64

In [63]:
sns.countplot(Data4['Plataforma'])

NameError: name 'sns' is not defined